client/basic-authorship/src/basic_authorship.rs

```diff
@@ -196,14 +196,23 @@ impl<A, B, Block, C> ProposerInner<B, Block, C, A>
 
 		// We don't check the API versions any further here since the dispatch compatibility
 		// check should be enough.
-		for extrinsic in self.client.runtime_api()
+		for inherent in self.client.runtime_api()
 			.inherent_extrinsics_with_context(
 				&self.parent_id,
 				ExecutionContext::BlockConstruction,
 				inherent_data
 			)?
 		{
-			block_builder.push(extrinsic)?;
+			match block_builder.push(inherent) {
```

Comment:
Don't we need to abort when we see a `BadMandatory` here?

Reply:
Do we? I'm presuming that it's probably better to attempt to produce a block without an (possibly needed) inherent than it is to avoid producing the block at all and risk the chain stalling.

Reply:
I'll put logic in to do this anyway since - assuming the dispatch is actually mandatory - it doesn't make much of a difference whether we fail here or fail when we try to finish up the block.